consensus/poet/core/sawtooth_poet/poet_consensus/utils.py

```diff
@@ -370,15 +362,348 @@ def get_consensus_state_for_block_id(
+
+    Returns:
+        True if allowing the validator to claim the block would result in the
+        validator being allowed to claim more frequently than statistically
+        expected, False otherwise
+    """
+    # If there are note enough blocks in the block chain to apply the zTest
+    # (i.e., we have not progressed past the blocks for which the local mean
+    # is calculated as a fixed ratio of the target to initial wait), simply
+    # short-circuit the test an allow the block to be claimed.
+    if consensus_state.total_block_claim_count < \
+            poet_config_view.fixed_duration_block_count:
+        return False
+
+    # Build up the population estimate list for the block chain and then add
+    # the new information (i.e., the validator trying to claim as well as the
+    # population estimate) to the front to maintain the order of most-recent
+    # to least-recent.
+    population_estimate_list = \
+        _build_population_estimate_list(
+            block_id=previous_block_id,
+            consensus_state=consensus_state,
+            poet_config_view=poet_config_view,
+            block_cache=block_cache,
+            poet_enclave_module=poet_enclave_module)
+    population_estimate_list.appendleft(
+        _EstimateInfo(
+            population_estimate=population_estimate,
+            previous_block_id=previous_block_id,
+            validator_id=validator_info.id))
+
+    observed_wins = 0
+    expected_wins = 0
+    block_count = 0
+    minimum_win_count = poet_config_view.ztest_minimum_win_count
+    maximum_win_deviation = poet_config_view.ztest_maximum_win_deviation
+
+    # We are now going to compute a "1 sample Z test" for each
+    # progressive range of results in the history.  Test the
+    # hypothesis that validator won elections (i.e., was able to
+    # claim blocks) with higher mean that expected.
+    #
+    # See: http://www.cogsci.ucsd.edu/classes/SP07/COGS14/NOTES/
+    #             binomial_ztest.pdf
+
+    for estimate_info in population_estimate_list:
+        # Keep track of the number of blocks and the expected number of wins
+        # up to this point.
+        block_count += 1
+        expected_wins += 1.0 / estimate_info.population_estimate
+
+        # If the validator trying to claim the block also claimed this block,
+        # update the number of blocks won and if we have seen more than the
+        # number of wins necessary to trigger the zTest, then we are going to
+        # figure out if the validator is winning too frequently.
+        if estimate_info.validator_id == validator_info.id:
+            observed_wins += 1
+            if observed_wins > minimum_win_count and \
+                    observed_wins > expected_wins:
+                probability = expected_wins / block_count
+                standard_deviation = \
+                    math.sqrt(block_count * probability *
+                              (1.0 - probability))
```

Comment:
If you prefer the expression as is for readability with your referenced z-test article that's fine.
Otherwise, consider block_count algebraically cancels so you have s=sqrt(E-EP)